File: Backend/src/main/java/com/tripnest/notification/service/NotificationService.java
package com.tripnest.notification.service;

import com.tripnest.notification.dto.NotificationDto;
import com.tripnest.notification.entity.NotificationType;

import java.util.List;

public interface NotificationService {
    void createNotification(String username, String title, String message, NotificationType type, String link);
    List<NotificationDto> getUserNotifications(String username);
    List<NotificationDto> getUnreadNotifications(String username);
    long getUnreadCount(String username);
    void markAsRead(Long notificationId, String username);
    void markAllAsRead(String username);
    void deleteNotification(Long notificationId, String username);
}

File: frontend/src/services/notificationService.js
import api from './apiClient';

export const notificationService = {
  getUserNotifications: async () => {
    return await api.get('/notifications');
  },

  getUnreadNotifications: async () => {
    return await api.get('/notifications/unread');
  },

  getUnreadCount: async () => {
    return await api.get('/notifications/unread/count');
  },

  markAsRead: async (id) => {
    return await api.put(`/notifications/${id}/read`);
  },

  markAllAsRead: async () => {
    return await api.put('/notifications/read-all');
  },

  deleteNotification: async (id) => {
    return await api.delete(`/notifications/${id}`);
  }
};
